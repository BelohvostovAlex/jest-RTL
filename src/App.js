import React from "react";

function App() {
  const [data, setData] = React.useState(null)
  
  React.useEffect(() => {
    setTimeout(() => setData({name: 'alex'}), 1000)
  }, [])

  return (
    <div className="App">
      {data && <div>data</div>}
      <h1>Hello world</h1>
      <button>click</button>
      <input type="text" placeholder="type ur username..."/>
    </div>
  );
}

export default App;
