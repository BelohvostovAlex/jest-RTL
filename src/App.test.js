import { render, screen } from '@testing-library/react';
import App from './App';

describe('testing app component', () => {
  test('renders learn react link', () => {
    render(<App />);
    const helloWorldElem = screen.getByText(/hello world/i);
    const helloAllElem = screen.queryByText(/hello all/i);
    const btn = screen.getByRole('button')
    const input = screen.getByPlaceholderText(/type ur username/i)
 
    expect(helloWorldElem).toBeInTheDocument();
    expect(helloAllElem).toBeNull();
    expect(btn).toBeInTheDocument();
    expect(input).toBeInTheDocument();

  });
  test('async div appearence', async () => {
    render(<App />);
    screen.debug()
    const helloElem = await screen.findByText(/data/i)
    expect(helloElem).toBeInTheDocument();
    screen.debug()
  })
})
