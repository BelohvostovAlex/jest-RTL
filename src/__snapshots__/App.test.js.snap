// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing app component renders learn react link 1`] = `
<input
  placeholder="type ur username..."
  type="text"
/>
`;
